usr/share/rear/lib/bootloader-functions.sh

```diff
@@ -547,11 +547,16 @@
     test "$grub2_initrd" || BugError "create_grub2_cfg function called without grub2_initrd argument"
     DebugPrint "Configuring GRUB2 initrd $grub2_initrd"
     local grub2_search_root_command="$3"
+    # Because GRUB2_SEARCH_ROOT_COMMAND is empty in default.conf
+    # we can provide final power to the user and always respect
+    # GRUB2_SEARCH_ROOT_COMMAND when the user has set it
+    # cf. https://github.com/rear/rear/issues/3024#issuecomment-1630963778
     if ! test "$grub2_search_root_command" ; then
-        test "$grub2_set_root" && grub2_search_root_command="set root=$grub2_set_root"
+        test "$GRUB2_SEARCH_ROOT_COMMAND" && grub2_search_root_command="$GRUB2_SEARCH_ROOT_COMMAND"
```

Comment:
Just a thought:

I agree with the check that having the same variable in all caps and in lowercase is at least confusing (although the check is in general wrong as it doesn't know about `default.conf`).

Is that something we can rectify or would that be too much of a change? This file is hard to read in any case, so maybe this is the least of all problems.

@jsmeix you spend more time here than me so I'll defer to your judgement.

Reply:
Those lowercase named variables are local variables that
intentionally match uppercase named user config variables.
Those lowercase named variables are meant to be used
inside a function or inside a script
when the value could be modified for that local usage
without modifying the user config variable.

I do not like it when user config variables get modified.

I think normally user config variables should be kept as is
so that different scripts could behave according to what
the user had specified and not according to what a previous
script may have changed from what the user had specified.

Probably I am thinking too complicated and I should
just use user config variables like any other variables.

The whole create_grub2_cfg() usage is such a mess
which at least I cannot maintain with reasonable effort
so that I may even clean that up mecilessly right now
and afterwards I can fix what I may have done wrong.

Reply:
I won't do a general cleanup via this pull request,
cf. https://github.com/rear/rear/pull/3025#issuecomment-1635340530

Reply:
The lower case variables grub2_set_root_command
and grub2_search_root_command are no longer needed, see
https://github.com/rear/rear/pull/3025#issuecomment-1638156038